### ISSUE TITLE
Play Protect certification shows Device is certified, but McDonalds app still detects root?

### ISSUE THREAD

**Issue by chaoscreater** [x1337cn/AutoPIF-Next#1]
Hi there, 

I've had PlayIntegrityNext and McDonalds app working a week ago and now it's broken. I've updated to the latest PlayIntegrity Next and got what I think is a working PIF.json. The problem is that I get "Meet_Device_Integrity" only sometimes, not all the time. I've tried multiple PIFs already, same issue. My Play Protect certification shows Device is certified, so it should be working correctly. But McDonalds app still thinks device is rooted. I've also tried using chiteroman's PlayIntegrityFix latest module, same issue.

I've tried yours too and I can get the Play Protect certification to show Device is certified, but I still can't get McDonalds to work.

I'm on Kitsune Mask canary build 26404. Not using shamiko.

Have cleared Google Play, Google Play Services and Google Services Framework already as well, multiple times.

Any ideas what I'm doing wrong? Any help would be much appreciated.

**Comment by x1337cn** [comment]
Try to uninstall and install McDonald's app also try to use HMA (Hide My Applist) LSposed Module it should work since the device passed the device integrity it should properly work.

**Comment by chaoscreater** [comment]
Thanks for your reply. I've already tried to uninstall and reinstall McDonald app, even tried to downgrade to an older version. Didn't work.

Have already been using HMA and it works well for other apps like banking, etc. But it doesn't seem to work for McDonalds.

Also tried the latest Magisk Alpha and Magisk Delta (Kitsune Mask) with Shamiko (latest version). They all can't pass McDonalds root check.

The problem is that device integrity is not passing consistently every time. It only passes on every 2nd or 3rd check.

**Comment by chaoscreater** [comment]
Got it working.



Turns out that it's really something to do with the xiaomi.eu ROM itself. Even though I disabled the xiaomi.eu injection module (as it conflicts with PlayIntegrity Fix modules), there's still something going on in the background it seems. When you flash the PlayIntegrity modules, they actually have a .replace file for the xiaomi.eu injection module and perhaps it's not working correctly.



Anyway, I found that there's actually a dedicated thread for the xiaomi.eu injection module:

https://xiaomi.eu/community/threads/xiaomieu-module-app.70890/#post-714756



I've uninstalled the PlayIntegrity module and enabled my xiaomi.eu injection module and updated it. After that, device integrity passes every time and McDonalds app work.



When I was using the PlayInjection modules, device integrity would pass only every 2nd or 3rd try. Play Store would show device is certified, but McDonalds app wouldn't work because device integrity wasn't passing 100% consistently.
